### PR TITLE
Update KeyVault to enable live testing in sovereign clouds for multiple services

### PR DESCRIPTION
**Pull request by v-hongli1**
These changes enable KeyVault to run live tests against Public, UsGov and China.

1. Update file: test-resources.json
i) Match with the `ArmTemplateParameter` setting, so change the parameter name of the endpoint suffix to `keyVaultEndpointSuffix`.
ii) Modified permission of access policies for fix `The user, group or application 'appid=***;oid=***;iss=https://sts.windows.net/***/' does not have keys getrotationpolicy permission on key vault 't1e479d51bbe44e7d;location=***'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287` excception.
iii) Fix the execution exception of `AccessTokenUtilTest.testGetAuthorizationToken` in UsGov and China Cloud, add output parameter `KEY_VAULT_ENDPOINT_SUFFIX` 

2. Update file: /keys/KeyClientTestBase.java
According to the repair suggestions in [issue#18142](https://github.com/Azure/azure-sdk-for-js/issues/18142), add the cloud test platform judgment method `isPublicCloud` .

3. Update file: /keys/KeyClientTest.java, /keys/KeyAsyncClientTest.java
According to the repair suggestions in [issue#18142](https://github.com/Azure/azure-sdk-for-js/issues/18142), skip Key Rotation related test cases in UsGov and China cloud.

4. Update file: /jca/AccessTokenUtilTest.java
Fix the execution exception of `AccessTokenUtilTest.testGetAuthorizationToken` in UsGov and China Cloud, and modify the parameter setting of `AccessTokenUtil.getAccessToken`. 

5. Update file: /certificates/CertificateClientTest.java
i) Fixed the `org.opentest4j.AssertionFailedError: expected: <false> but was: <true>` exception in UsGov and China cloud. 
ii) Adjust the process flow to avoid `waitForCompletion()` to interfere with result data interference 

6. In the pipeline test, system parameters such as `AZURE_KEYVAULT_CERTIFICATE_NAME` are stored in `Configuration`, and cannot get them by `System.getenv(name)`, so replaced with `Configuration.getGlobalConfiguration().get(name, default Value)`,  and the  `defaultValue` is obtained through `System.getenv(name)` system parameters or defined constants.
e.g. 
Update `certificateName = System.getenv("AZURE_KEYVAULT_CERTIFICATE_NAME")` 
to `certificateName = Configuration.getGlobalConfiguration().get("AZURE_KEYVAULT_CERTIFICATE_NAME", System.getenv("AZURE_KEYVAULT_CERTIFICATE_NAME"))`

Pipeline results:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1227309&view=results

@benbp, @scottaddie, @joshfree, @AlexGhiondea and @vcolin7 for notification.

### CODE REVIEW

**Review by vcolin7 — APPROVED** [review]
The changes look good to me, but I would wait for @chenrujun to review as well just to be sure, as he knows more about the JCA module than I do.

File: sdk/keyvault/test-resources.json
Comment (chenrujun) [inline sdk/keyvault/test-resources.json:79]: > Match with the ArmTemplateParameter setting, so change the parameter name of the endpoint suffix to keyVaultEndpointSuffix.

Could you please where to find document about `ArmTemplateParameter`?

File: sdk/keyvault/test-resources.json
Comment (chenrujun) [inline sdk/keyvault/test-resources.json:70]: Why lower case is necessary?

File: sdk/keyvault/test-resources.json
Comment (chenrujun) [inline sdk/keyvault/test-resources.json:209]: Same here, why upper case is necessary?

File: sdk/keyvault/test-resources.json
Comment (chenrujun) [inline sdk/keyvault/test-resources.json:210]: Could you please keep original order? That's alphabet order.

File: sdk/keyvault/tests.yml
Comment (chenrujun) [inline sdk/keyvault/tests.yml:14]: Why it's necessary to change location?

File: sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/PropertyConvertorUtils.java
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/PropertyConvertorUtils.java:22]: How about adding a new method in `PropertyConvertorUtils`:
```
    public static final String getPropertyValue(String property) {
        return Configuration.getGlobalConfiguration().get(property, System.getenv(property));
    }
```

And use it in all places?

File: sdk/keyvault/test-resources.json
Comment (hongli750210) [reply sdk/keyvault/test-resources.json:79]: Please refer to is[sue#23646 ](https://github.com/Azure/azure-sdk-for-java/issues/23646)

File: sdk/keyvault/test-resources.json
Comment (hongli750210) [reply sdk/keyvault/test-resources.json:209]: Due to the incomplete setting of KeyVault permissions, it is not possible to perform rotation strategy related tests in UsGov and China Cloud. The revised settings are taken from the template data exported from the Azure portal.

File: sdk/keyvault/test-resources.json
Comment (hongli750210) [reply sdk/keyvault/test-resources.json:210]: Due to the incomplete setting of KeyVault permissions, it is not possible to perform rotation strategy related tests in UsGov and China Cloud. The revised settings are taken from the template data exported from the Azure portal.

File: sdk/keyvault/tests.yml
Comment (hongli750210) [reply sdk/keyvault/tests.yml:14]: When testing in Canary Cloud, a deploy resource exception occurred. According to the description of the error message, it was judged that the location setting was incorrect.

File: sdk/keyvault/test-resources.json
Comment (hongli750210) [reply sdk/keyvault/test-resources.json:70]: The deploy resource exception occurred in all cloud tests before fixed.

File: sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/PropertyConvertorUtils.java
Comment (hongli750210) [reply sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/PropertyConvertorUtils.java:22]: Fixed in new version.

File: sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/AccessTokenUtilTest.java
Comment (hongli750210) [reply sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/AccessTokenUtilTest.java:42]: Fixed in the new version.

**Review by chenrujun — APPROVED** [review]
LGTM.

File: sdk/keyvault/tests.yml
Comment (hongli750210) [reply sdk/keyvault/tests.yml]: Okay, the `Canary` test entry has been completely deleted in new version.

File: sdk/keyvault/tests.yml
Comment (benbp) [inline sdk/keyvault/tests.yml:9]: This is unnecessary as we already default to LIVE in the live tests (https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/templates/stages/archetype-sdk-tests.yml#L59-L61).

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java
Comment (benbp) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java:582]: Is there a way to mark these tests as skipped rather than showing that they pass with empty operation? I see the use of `Assumptions.assumeTrue` below, perhaps that's a better option, but I'll let a java sdk dev weigh in.

File: sdk/keyvault/tests.yml
Comment (hongli750210) [reply sdk/keyvault/tests.yml]: Fixed in the new version.

**Comment by hongli750210** [comment]
@vcolin7  I rerun a new weekly pipeline, but the `KeyVaultErrorException` occurred when the test cases of `KeyVaultBackupAsyncClientTest` and `KeyVaultBackupClientTest` were executed on OS `ubuntu2004_TestFromSource_HSM_nettty_111` of Public cloud.
For more details please check [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1256782&view=logs&j=a17b4ad0-a586-5af9-f529-32048a166ec8&t=72b63d00-91e1-520f-2d54-361694966d16).

**Pull request merged**
